### PR TITLE
do not render KMS encrypted content via go templates

### PR DESCRIPTION
**Pull request by xh3b4sd**
## Checklist

- [x] Update changelog in CHANGELOG.md.

## Context 

Fixes https://github.com/giantswarm/giantswarm/issues/11517. It looks like so far we took the KMS encrypted apiserver encryption key and processed it in the golang templating engine for no good reason. This approach apparently fails randomly in case the KMS encrypted content sequence represents character sets invalid to the golang templating engine. Talking to @tfussell it appears to be best when handling the apiserver encryption key like the encrypted certificates which we only base64 encode instead of decrypting again. Note that we struggle with this issue for about 2 years now and never could figure a legitimate fix.

### CODE REVIEW

**Comment by xh3b4sd** [comment]
Still need to test this in the wild.

**Comment by xh3b4sd** [comment]
Tested on `gauss`. 

```
$ k get node
NAME                                           STATUS   ROLES    AGE   VERSION
ip-10-1-18-120.eu-central-1.compute.internal   Ready    master   17m   v1.16.9
ip-10-1-18-179.eu-central-1.compute.internal   Ready    master   16m   v1.16.9
ip-10-1-18-37.eu-central-1.compute.internal    Ready    master   17m   v1.16.9
ip-10-1-19-100.eu-central-1.compute.internal   Ready    worker   17m   v1.16.9
ip-10-1-19-159.eu-central-1.compute.internal   Ready    worker   17m   v1.16.9
ip-10-1-19-31.eu-central-1.compute.internal    Ready    worker   17m   v1.16.9
```

File: service/internal/cloudconfig/tccpn_extension.go [hunk 1]
```diff
@@ -230,6 +217,24 @@ func (e *TCCPNExtension) Files() ([]k8scloudconfig.FileAsset, error) {
 
 	certsMeta := []k8scloudconfig.FileMetadata{}
 	{
+		{
+			meta := k8scloudconfig.FileMetadata{
```
Comment (tfussell) [inline service/internal/cloudconfig/tccpn_extension.go:221]:
```suggestion
			// Add to certsMeta slice so the encrypted encryption config file isn't passed through
			// k8scloudconfig.RenderFileAssetContent like other files in filesMeta.
			meta := k8scloudconfig.FileMetadata{
```

File: pkg/project/project.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ var (
 	gitSHA             = "n/a"
 	name        string = "aws-operator"
 	source      string = "https://github.com/giantswarm/aws-operator"
-	version            = "8.7.1-dev"
+	version            = "8.7.1-xh3b4sd"
```
Comment (tfussell) [inline pkg/project/project.go:8]:
```suggestion
	version            = "8.7.1-dev"
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -26,6 +26,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Fix cluster deletion issues in AWS using `DependsOn`.
 - Fix calico-policy only metrics endpoint.
 - Fix race condition in IPAM locking when lock already acquired.
+- Fix failing go template rendering of KMS encryption content.
```
Comment (tfussell) [inline CHANGELOG.md:29]: This should go under `Unreleased` I think.

Comment (xh3b4sd) [reply CHANGELOG.md]: fixed

**Pull request merged**